python/ray/data/_internal/execution/interfaces.py

```diff
@@ -70,21 +70,77 @@ def destroy_if_owned(self) -> int:
         return self.size_bytes() if should_free else 0
 
 
+@dataclass
+class ExecutionResources:
+    """Specifies resources usage or resource limits for execution.
+
+    The value `None` represents unknown resource usage or an unspecified limit.
+    """
+
+    # CPU usage in cores (Ray logical CPU slots).
+    cpu: Optional[float] = None
+
+    # GPU usage in devices (Ray logical GPU slots).
+    gpu: Optional[float] = None
+
+    # Object store memory usage in bytes.
+    object_store_memory: Optional[int] = None
+
+    def object_store_memory_str(self) -> str:
+        """Returns a human-readable string for the object store memory field."""
+        if self.object_store_memory is None:
+            return "None"
+        elif self.object_store_memory >= 1024 * 1024 * 1024:
+            return f"{round(self.object_store_memory / (1024 * 1024 * 1024), 2)} GiB"
+        else:
+            return f"{round(self.object_store_memory / (1024 * 1024), 2)} MiB"
+
+    def add(self, other: "ExecutionResources") -> "ExecutionResources":
+        """Adds execution resources.
+
+        Returns:
+            A new ExecutionResource object with summed resources.
+        """
+        total = ExecutionResources()
+        if self.cpu is not None or other.cpu is not None:
+            total.cpu = (self.cpu or 0.0) + (other.cpu or 0.0)
+        if self.gpu is not None or other.gpu is not None:
+            total.gpu = (self.gpu or 0.0) + (other.gpu or 0.0)
+        if (
+            self.object_store_memory is not None
+            or other.object_store_memory is not None
+        ):
+            total.object_store_memory = (self.object_store_memory or 0.0) + (
+                other.object_store_memory or 0.0
+            )
+        return total
+
+    def satisfies_limits(self, limit: "ExecutionResources") -> bool:
```

Comment:
super nit: should we stick with `satisfy_limits` given `add()` above?

Reply:
What do you mean?

Reply:
sorry for confusion, but should we stick with either (`satisfy_limits()`, `add()`), or (`satisfies_limits()`, `adds()`)? These two methods naming look inconsistent to me. But it's a minor point.

Reply:
Oh, I can remove the s.

Reply:
thanks!